### PR TITLE
[ECO-278] Remove pdoc build artifacts

### PR DESCRIPTION
**Pull request by alnoki**
Presently, the pdoc build html is included in the repo.

Docs should be built through CI, and artifacts should be ignored from the repo.

This PR removes build artifacts and cleans up some of the configs/README content associated with the Python SDK.

### CODE REVIEW

File: src/python/sdk/README.md
Comment (bogberry) [inline src/python/sdk/README.md:1]: If this is going to be on the main page of the pdoc, its contents should be aimed at people looking to use the SDK, not people looking to build the documentation.

File: .gitignore [hunk 1]
```diff
@@ -20,6 +20,7 @@ doc/litepaper/main.pdf
 # Python
 *.pyc
 *.egg-info
+src/python/sdk/docs/*
```
Comment (bogberry) [inline .gitignore:23]: Prefer per-project gitignores. This repo contains a lot of different languages, all with different files that should be ignored.

.gitignore files in subdirectories also prevents issues from arising when directory structure changes

File: src/python/sdk/pyproject.toml [hunk 2]
```diff
@@ -10,6 +10,7 @@ packages = [{ include = "econia_sdk" }]
 aptos-sdk = "^0.5.1"
 python = "^3.8"
 autoflake = "^2.2.0"
+pdoc = "^14.0.0"
```
Comment (bogberry) [inline src/python/sdk/pyproject.toml:13]: Use Poetry dependency groups:
<https://python-poetry.org/docs/managing-dependencies/>

e.g. `poetry add --group docs pdoc`

**Review by bogberry — CHANGES_REQUESTED** [review]

File: src/python/sdk/README.md
Comment (alnoki) [reply src/python/sdk/README.md:1]: @bogberry addressed in bea4ab5

File: .gitignore (hunk 1 above)
Comment (alnoki) [reply .gitignore]: @bogberry addressed in bea4ab5

See also https://linear.app/econia-labs/issue/ECO-281/enforce-granular-gitignores

File: src/python/sdk/pyproject.toml (hunk 2 above)
Comment (alnoki) [reply src/python/sdk/pyproject.toml]: @bogberry addressed in bea4ab5

File: src/python/sdk/README.md
Comment (bogberry) [reply src/python/sdk/README.md:1]: I think we should be giving users a little more than this when they visit the main page of the Python SDK documentation.

Screenshot 2023-08-01 at 9 51 56

Comment (alnoki) [reply src/python/sdk/README.md:1]: @bogberry https://linear.app/econia-labs/issue/ECO-323/provide-more-information-on-python-sdk-readme

Comment (bogberry) [reply src/python/sdk/README.md:1]: Cool

**Review by bogberry — APPROVED** [review]

**Pull request merged**